### ISSUE TITLE
Line break indentation (double indentation) isn't possible with tab key by default

### ISSUE THREAD

**Issue by clangdo** [godotengine/emacs-gdscript-mode#92]
## Versions
Emacs 27.1
gdscript-mode 1.3.0
Godot 3.2.3 (Not that it should matter)

## Behaviors
(Note the following code blocks use spaces as they are easier to insert using github, but I have not changed any default settings in my editor, so I'm using tabs.)
### Observed
When breaking a line I am unable to use tab an additional time to differentiate the broken line from the block it begins:
```gdscript
if(boolean_one and boolean_two and 
    boolean_three and boolean_four):
    do_something()
```
Instead the tab cycles between the previous levels of indentation.
### Expected
When breaking a line I should be able to tab in an additional time, as recommended by the style guide:
```gdscript
if(boolean_one and boolean_two and 
        boolean_three and boolean_four):
    do_something()
```

## Additional Information
The relevant style guide section is [here](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#indentation)

For what it's worth I don't have any custom indentation settings in my emacs initialization elisp. Furthermore gdscript-mode is the only installed package.

**Comment by NathanLovato** [comment]
Mh, I can do it here with tab, which is bound to `indent-for-tab-command`, by pressing tab three times. The first press moves the hanging line back with the upper line but then it indents forward.

Looking at our indent functions, taken from Python support, it seems it's designed to do that. Which work in some cases but not for hanging indents:

```lisp
(defun gdscript-indent-line-function ()
  "`indent-line-function' for Gdscript mode.
When the variable `last-command' is equal to one of the symbols
inside `gdscript-indent-trigger-commands' it cycles possible
indentation levels from right to left."
  (gdscript-indent-line
   (and (memq this-command gdscript-indent-trigger-commands)
        (eq last-command this-command))))
```

**Comment by clangdo** [comment]
I will go through some troubleshooting steps to see if something is wrong on my end. For me, when I press <tab> multiple times, it just cycles through the levels of dedentation, and never jumps past the previous level like you describe. I'm a total newbie when it comes to elisp, but it's high time I learned, so I will work on understanding why I'm seeing the behavior I'm seeing and hopefully that will help.

**Comment by NathanLovato** [comment]
Thanks, I have a lot on my plate at the moment and limited time to work on this repo. You can find the corresponding code in `gdscript-indent.el` and `gdscript-indent-line-function`, plus all the functions referenced there like `gdscript-indent-line`.

Lisp has a weird syntax at first but once you get the hang of it, the tools to evaluate any piece of code while running emacs are really cool

**Comment by clangdo** [comment]
I've finally got around to doing some work on this. As far as I can tell, (still working with 1.3.0 here), the default python indentation does the following when it's in a line continuation:

https://github.com/godotengine/emacs-gdscript-mode/blob/099ca4471fe5e44da57f1d34e3d3b5604e619977/gdscript-indent-and-nav.el#L281-L286

Basically it assumes that you'd like to continue the line at `start` which in this context (`:inside-paren`) happens to be the opening parenthesis' position. Note that this doesn't run contrary to PEP8, which allows this as one way of continuing lines.
(see https://www.python.org/dev/peps/pep-0008/#indentation). That said, this does run contrary to the recommendation for gdscript, which is more specific, and to which a link can be found in my original post.

I patched it to the following and it seems to work perfectly for me (at least within parentheses).
```elisp
(`(,(or :inside-paren
              :after-backslash-block-continuation
              :after-backslash-dotted-continuation) . ,start)
       ;; Use the column given by the context.
       (goto-char start)
       (+ (current-indentation) (* 2 gdscript-indent-offset)))```

Since you cant replicate the issue I haven't created a pull request, but let me know if you would like me to. (Version `1.4.0` seems to be the same at first glance.)

**Comment by NathanLovato** [comment]
Sure, a PR would be most welcome!

Not everyone likes the double hanging identation, it tends to push code a lot to the right. I'd make that `* 2` optional (a new defcustom?). Actually, I think it'll conflict with the GDScript formatter, which uses one tab inside hanging parens.

**Comment by clangdo** [comment]
Indeed, code style is a matter of opinion and what fits each team best. I'm all for including a defcustom for this, but my question becomes what should it default to? If it defaults to one tab on hanging indent then we aren't following the Godot docs. If it defaults to two then we aren't agreeing with the GDScript format folks. Anyone else want to weigh in? Perhaps I should start a discussion in a more visible part of the community so more people have a say? (Perhaps a discussion in https://github.com/godotengine/godot-docs is warranted?)

**Comment by clangdo** [comment]
I think as a preliminary I will create a separate variable `gdscript-indent-line-continuation-scale` and create a pull request, this gives users the ability to change `gdscript-indent-def-block-scale` independently. After I create a pull request we can fix any newbie mistakes I may make.

**Comment by clangdo** [comment]
I have a pr [here](https://github.com/godotengine/emacs-gdscript-mode/pull/100) that could close this issue when it's approved.

**Comment by NathanLovato** [comment]
Fixed with #100
